test/azul_test_case.py

```diff
@@ -29,36 +32,42 @@ class AzulTestCase(TestCase):
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
+
+        class Pat(str):
+            pass
+
         catch_warnings = warnings.catch_warnings(record=True)
         # Use tuple assignment to modify state atomically
         cls._catch_warnings, cls._caught_warnings = catch_warnings, catch_warnings.__enter__()
         permitted_warnings_ = {
             ResourceWarning: {
-                r'.*<ssl.SSLSocket.*>',
-                r'.*<socket.socket.*>'
+                Pat(r'.*<ssl.SSLSocket.*>'),
+                Pat(r'.*<socket.socket.*>')
             },
             DeprecationWarning: {
-                r'Call to deprecated method .*\. \(DOS support will be removed\)',
-                '.*Call to deprecated method fetch_bundle_manifest.*',
+                Pat(r'Call to deprecated method .*\. \(DOS support will be removed\)'),
+                'Call to deprecated method fetch_bundle_manifest',
                 'ProjectContact.contact_name is deprecated',
                 'File.file_format is deprecated',
                 'ProjectPublication.publication_title is deprecated',
                 'ProjectPublication.publication_url is deprecated',
                 'CellLine.cell_line_type is deprecated',
-                '.*humancellatlas.data.metadata.api.DissociationProcess',
-                '.*humancellatlas.data.metadata.api.EnrichmentProcess',
-                '.+humancellatlas.data.metadata.api.LibraryPreparationProcess',
-                '.*humancellatlas.data.metadata.api.SequencingProcess',
+                'humancellatlas.data.metadata.api.DissociationProcess',
+                Pat(r'.*humancellatlas\.data\.metadata\.api\.EnrichmentProcess'),
+                Pat(r'.+humancellatlas\.data\.metadata\.api\.LibraryPreparationProcess'),
+                Pat(r'.*humancellatlas\.data\.metadata\.api\.SequencingProcess'),
                 # FIXME: Upgrade tenacity
                 #        https://github.com/DataBiosphere/azul/issues/2070
-                r'"@coroutine" decorator is deprecated since Python 3\.8, use "async def" instead'
+                '"@coroutine" decorator is deprecated since Python 3.8, use "async def" instead'
             },
             UserWarning: {
-                r'https://github\.com/DataBiosphere/azul/issues/2114',
+                'https://github.com/DataBiosphere/azul/issues/2114',
             }
         }
         for warning_class, message_patterns in permitted_warnings_.items():
             for message_pattern in message_patterns:
+                if type(message_pattern) is not Pat:
```

Comment:
I'm not sure if there's anything wrong with checking a type this way, but the common pattern---both in our code and in my experiences with Python---is to use `isinstance()` instead. 
```suggestion
                if not isinstance(message_pattern, Pat):
```

Reply:
good point, ill change this, also https://stackoverflow.com/a/1549854/2205587